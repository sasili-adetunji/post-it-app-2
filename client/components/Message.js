import React from 'react';
import mui from 'material-ui';
import API from '../Api';
import { ListItem } from 'material-ui/List';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';


class Message extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <ListItem>
          {this.props.message.messageText} - {this.props.message.isRead}
        </ListItem>
      </div>
    );
  }
}

export default Message;
