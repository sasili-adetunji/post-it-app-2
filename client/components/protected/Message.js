import React from 'react';

/**
 *
 *
 * @class Message
 * @extends {React.Component}
 */
class Message extends React.Component {
  constructor(props) {
    super(props);
  }
  /**
   *
   * renders the message components
   * @returns {void}
   * @memberof Message
   */
  render() {
    return (
      <div className="panel-body msg_container_base">
        <div className="row msg_container base_sent">
          <div className="col-md-10 col-xs-10">
            <ul>
              <div className="messages msg_sent">
                <p> { this.props.message.messageText } </p>
                <time dateTime="2009-11-13T20:00">Timothy â€¢ 51 min</time>
              </div>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}
export default Message;
