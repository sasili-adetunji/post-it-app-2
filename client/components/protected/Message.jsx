import React from 'react';

const Message = ({ message }) => (
  <div className="message">
    <div className="row">
      <div className="col-md-10">
        <p> <b> {message.author} </b> </p>
      </div>
      <div className="col-md-2" id="priorityDisplay">
        <p><i> {message.priorityLevel} </i></p>
      </div>
    </div>
    <div className="row">
      <div className="col-md-12">
        <p> {message.messageText}</p>
      </div>
    </div>
    <div className="row">
      <div className="col-md-10">
        <p> {message.date}</p>
      </div>
      <div className="col-md-2" />
    </div>
  </div>
    );
export default Message;
