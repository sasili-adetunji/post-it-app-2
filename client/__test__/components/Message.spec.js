import React from 'react';
import { shallow, mount } from 'enzyme';
import Message from '../../components/protected/Message.jsx';
import {JSDOM} from 'jsdom';

const jsdom = new JSDOM('<!doctype html><html><body></body></html>');

const { window } = jsdom;
global.window = window;
global.document = window.document;


describe('User components', () => {
  const props = {
    message: {
    messageText: 'sasil',
    date: '',
    status: 'Read',
    author: 'Wash',
    },
  readUser: {
    name: 'Sasili',
    name: 'Qudus'
  }
  };
const shallowComponent = () => shallow(<Message {...props} />)
const component = () => mount(<Message {...props} />)

  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain four div', () => {
    expect(shallowComponent().find('div').length).toEqual(4);
  });
  it('Should contain one ul', () => {
    expect(shallowComponent().find('ul').length).toEqual(1);
  });
  it('Should contain one p', () => {
    expect(shallowComponent().find('p').length).toEqual(1);
  });
  it('should recieve props', () => {
    expect(Object.keys(component().props()).length).toBeGreaterThan(0);
  });
});
