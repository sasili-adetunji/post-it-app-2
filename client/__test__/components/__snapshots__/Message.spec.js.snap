// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User components should match snapshot test 1`] = `[Function]`;
